lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -788,15 +791,24 @@ impl LayerModel {
         if self.flags.contains(LayerFlags::INHERIT_PARENT_CAMERA) {
             self.set_camera(parent.camera());
         }
+        // Depth order may be affected by new parent's `global_element_depth_order`.
+        self.depth_order_dirty.set();
```

Comment:
"may be affected" sounds like we are not sure about it. It's really hard to check the logic here, so I've got a question - are we setting dirty flags ONLY when necessarily, or are we setting them in more cases than needed? So far, the display object algorithm was always designed to set the minimum number of dirty flags and make as small amount of computations as possible and I think we should always try to do so, as this is foundation for almost everything visible on the screen in Enso.

Reply:
Oh, I confused this file with `display::Object` ... That's why I told its foundation to everything visible on the screen. Dirty flags in layers should have way smaller effect on performance. Anyway, thanks for the explanation and improved docs!